plugins/modules/zos_stat.py

```diff
@@ -0,0 +1,1685 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2025
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+---
+module: zos_stat
+version_added: '1.14.0'
+author:
+  - "Ivan Moreno (@rexemin)"
+short_description: Retrieve facts from z/OS
+description:
+  - The L(zos_stat,./zos_stat.html) module retrieves facts from resources
+    stored in a z/OS system.
+  - Resources that can be queried are files, data sets and aggregates.
+options:
+  name:
+    description:
+        - Name of a data set or aggregate, or a file path, to query.
+        - Data sets can be sequential, partitioned (PDS), partitioned
+          extended (PDSE), VSAMs, generation data groups (GDG) or
+          generation data sets (GDS).
+    type: str
+    required: true
+  volume:
+    description:
+        - Name of the volume where the data set will be searched on.
+        - Required when getting facts from a data set. Ignored otherwise.
+    type: str
+    required: false
+  type:
+    description:
+        - Type of resource to query.
+    type: str
+    required: false
+    default: data_set
+    choices:
+      - data_set
+      - file
+      - aggregate
+  sms_managed:
+    description:
+        - Whether the data set is managed by the Storage Management Subsystem.
+        - It will cause the module to retrieve additional information, may
+          take longer to query all attributes of a data set.
+        - If the data set is a PDSE and the Ansible user has RACF READ authority
+          on it, retrieving SMS information will update the last referenced
+          date of the data set.
+        - If the system finds the data set is not actually managed by SMS, the
+          rest of the attributes will still be queried and this will be noted
+          in the output from the task.
+    type: bool
+    required: false
+    default: true
+  tmp_hlq:
+    description:
+      - Override the default high level qualifier (HLQ) for temporary data
+        sets.
+      - The default HLQ is the Ansible user used to execute the module and
+        if that is not available, then the environment variable value
+        C(TMPHLQ) is used.
+    type: str
+    required: false
+
+notes:
+  - When querying data sets, the module will create a temporary data set
+    that requires around 4 kilobytes of available space on the remote host.
```

Comment:
Changing this to be more ansible-like language.
```suggestion
    that requires around 4 kilobytes of available space on the managed node.
```